### PR TITLE
Polynomial API rust bindings documentation

### PR DESCRIPTION
**Pull request by yshekel**

### CODE REVIEW

File: docs/docs/icicle/rust-bindings/polynomials.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+
+## Initialization Requirements
+
+Before utilizing any functions from the polynomial API, it is mandatory to initialize the appropriate polynomial backend (e.g., CUDA). Additionally, the NTT (Number Theoretic Transform) domain must also be initialized, as the CUDA backend relies on this for certain operations. Failing to properly initialize these components can result in errors.
+
+:::note
+**Field-Specific Initialization Requirement**
+
+The Icicle library is structured such that each field or curve has its dedicated library implementation. As a result, initialization must be performed individually for each field or curve to ensure the correct setup and functionality of the library.
```
Comment (LeonHibnik) [inline docs/docs/icicle/rust-bindings/polynomials.md:19]:
```suggestion
The ICICLE library is structured such that each field or curve has its dedicated library implementation. As a result, initialization must be performed individually for each field or curve to ensure the correct setup and functionality of the library.
```

File: docs/docs/icicle/rust-bindings/polynomials.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+// For operations over another field, such as BN254
+use icicle_bn254::polynomials::DensePolynomial as PolynomialBn254;
+// Use PolynomialBn254 similarly
+```
+
+### Creation
+Polynomials can be created from coefficients or evaluations:
+
+```rust
+// Assume F is the field type
```
Comment (LeonHibnik) [inline docs/docs/icicle/rust-bindings/polynomials.md:157]: Where's F?

**Review by LeonHibnik — APPROVED** [review]
Looks great! left minor comments

File: docs/docs/icicle/rust-bindings/polynomials.md (hunk 2 above)
Comment (yshekel) [reply docs/docs/icicle/rust-bindings/polynomials.md:157]: it's the field. for example icicle_bn254::curve::ScalarField
added a comment

**Pull request merged**